docs/testing/README.md

```diff
@@ -0,0 +1,26 @@
+# Testing in LocalStack
+
+- [Integration Tests](integration-tests/README.md)
+- [Parity Testing](parity-testing/README.md)
+- [Multi-account and Multi-region Testing](multi-account-region-testing/README.md)
+- [Terraform Tests](terraform-tests/README.md)
+
+## Rules for stable tests
+
+Through experience, we encountered some guiding principles and rules when it comes to testing LocalStack.
+These aim to ensure a stable pipeline and keep flakes minimal. Any newly added test and feature should keep these in mind!
+
+| **ID** | **Rule** |
+|-------|-----------|
+| R01   | Inform code owners and/or test authors when a test is marked as flaky so that it can be fixed ASAP. |
+| R02   | Do not assume external dependencies (AWS resources, files, packages, images, licenses) are indefinitely available. |
+| R02a  | Where possible, add automatic creation of dependencies if they no longer exist (e.g., S3 buckets, roles). |
+| R02b  | If on-demand creation is not possible, make access failures clearly visible for further investigation. |
+| R02c  | Add mechanisms to avoid access failures caused by rate limiting. |
+| R03   | Instead of waiting for a set amount of time, wait only as long as needed (with a reasonable cap). |
```

Comment:
perhaps we should rephrase this to not wait a _set_ amount of time in general, but rather use event-based notification mechanisms (like using SQS queues to wait for lambda results) where possible, and otherwise use polling with reasonable timeouts?
perhaps also adding that this is specifically for asynchronous operations?

Reply:
This is what I was trying to say with that but couldn't properly express it concisely enough